src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/VariableDeclarationUsageDistanceCheck.java

```diff
@@ -814,13 +814,15 @@ private static DetailAST getFirstNodeInsideSwitchBlock(
                 new ArrayList<>();
 
         // Checking variable usage inside all CASE_GROUP and SWITCH_RULE ast's.
-        TokenUtil.forEachChild(block, currentNode.getType(), node -> {
-            final DetailAST lastNodeInCaseGroup =
-                node.getLastChild();
-            if (isChild(lastNodeInCaseGroup, variable)) {
-                variableUsageExpressions.add(lastNodeInCaseGroup);
-            }
-        });
+        if (currentNode != null) {
```

Comment:
lets rename currentNode to blockStatement, wedo not iterate over nodes, there is not "current".
https://docs.oracle.com/javase/specs/jls/se7/html/jls-14.html#jls-14.11

Reply:
To avoid such nulls, it might be reasonable to make method 
```
getVariableExpressions(block, variable){
Optional.ofNullable(block.findFirstToken(TokenTypes.CASE_GROUP))
            .orElse(() -> block.findFirstToken(TokenTypes.SWITCH_RULE))
. map(
TokenUtil.forEachChild(block, currentNode.getType(), node -> {
                final DetailAST lastNodeInCaseGroup =
                    node.getLastChild();
                if (isChild(lastNodeInCaseGroup, variable)) {
                    variableUsageExpressions.add(lastNodeInCaseGroup);
                }
            })
).
orElseGet( List.empty);
}
```
and combine bodies.


of make it
```
TokenUtil.forEachChild(block, List.of(TokenTypes.CASE_GROUP, TokenTypes.SWITCH_RULE), .......
```

Reply:
Done, created method `getVariableUsageExpressionsInsideSwitchBlock()` so we get rid of null check,

Reply:
Why not TokenUtil.forEachChild(block, List.of(TokenTypes.CASE_GROUP, TokenTypes.SWITCH_RULE), ....... ?

It more clear

Reply:
And it is what we actually need to

Reply:
TokenUtil.forEachChild takes in an int as a second parameter, not a list.

Reply:
Of course, there is the workaround to make two calls to `TokenUtil.forEachChild()` if that is what you wanna do. One for each token `CASE_GROUP` and `SWITCH_RULE`.

Reply:
All this code want to do is to iterate over each nested block of switch, what is know to be "case" or switchRule. And it probably missed `default` that is not good, but might different issue.
So we just need to iterate over all nodes that are case or switchRule.
So `TokenUtil.forEachChild(block, List.of(TokenTypes.CASE_GROUP, TokenTypes.SWITCH_RULE)` will do exactly what we need.

The only obstacles I can imagine is pitest, which might complain that there is no test for switch with case and switchRule in same time. If this is a case, ok to stay on `.map().or()`

Reply:
Done. Method now looking for both `CASE_GROUP` and `SWITCH_RULE` in the same switch block. Even though this is not possible in the Java language to have `CASE_GROUP` and `SWITCH_RULE` in the same switch expression, I think you just want to be on the safe side, am I correct?